### DISCUSSION TITLE
Return Instances Where Name Equals String

### DISCUSSION THREAD

**Discussion by erickkendall** [aws/aws-cli#8230] [Q&A]
I have the following which will list the name and instance id

aws ec2 describe-instances --region us-east-1 --profile profile --query 'Reservations[*].Instances[].[Tags[?Key==`Name`].Value | join(`, `, @), InstanceId |  join(`, `, to_array(to_string(@))) ]' --output table

How can I limit the results to only those servers who match some string - like [Tags[?Key==`Name`].Value == serverxyz

**Comment by aBurmeseDev** [comment]
Hi @erickkendall - thanks for reaching out. 

Can you please elaborate more on which value you'd like to filter, for example, instance name, owner tag etc..? A combination of `filters` and `query` option would be what you're looking for. Here are an example that I think could be relevant:

- To filter instances of the specified type and only display their instance IDs, Availability Zone, and the specified tag value
    - The following describe-instances examples display the instance ID, Availability Zone, and the value of the Name tag for instances that have a tag with the name tag-key, in table format.

```
aws ec2 describe-instances \
    --filters Name=tag-key,Values=Name \
    --query 'Reservations[*].Instances[*].{Instance:InstanceId,AZ:Placement.AvailabilityZone,Name:Tags[?Key==`Name`]|[0].Value}' \
    --output table
```

Also `--query` uses JMESPath syntax and here are examples: https://jmespath.org/examples.html

Hope that helps!
